src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/ReadyToRunCodegenCompilation.cs

```diff
@@ -744,8 +748,19 @@ void CompileOneMethod(DependencyNodeCore<NodeFactory> dependency)
                 {
                     using (PerfEventSource.StartStopEvents.JitMethodEvents())
                     {
-                        // Create only 1 CorInfoImpl per thread.
-                        // This allows SuperPMI to rely on non-reuse of handles in ObjectToHandle
+                        s_methodsCompiledPerThread++;
+
+                        if (_parallelism == 1)
+                        {
+                            // Create only 1 CorInfoImpl per thread if not using parallelism
+                            // This allows SuperPMI to rely on non-reuse of handles in ObjectToHandle
+                        }
+                        else
+                        {
+                            if ((s_methodsCompiledPerThread % 1000) == 0)
```

Comment:
Is this condition part of the stress mode or is it something we're going to keep in the product?

Reply:
Something like this is something we should keep in the product. We have had periodic complaints about this growing without bound, and simply cleaning up periodically isn't terrible. Every 1000 methods is probably a bit too much/not enough, and it really should be tied to the size of the data structures being held onto though.

Reply:
If we keep this in the product, can we add a comment here about why we're doing it?